### PR TITLE
File list columns are now sortable

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Added ability to sort some columns in the file lists by clicking on
them. The sort direction appears with an indicator.

## Related Issue
Fixes https://github.com/owncloud/phoenix/issues/1854

## How Has This Been Tested?
- manual test
- acceptance tests

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] restore or keep sort order per file list (ex: sort by share time in "shared with others" then go back to "all files", no sort is set now)
- [x] add sortable columns in more lists (trashbin, shares)
- [x] add acceptance tests

### CODE REVIEW

**Comment by PVince81** [comment]
Now thinking, it won't be easy to have a sort field per table because the sorting is happening in the global state. Would require moving the global state's `activeFiles` into the object, and possibly more of the mixins. For this task I'll try to limit the changes and just make sure it doesn't behave too badly when switching between views. Maybe it will simply fallback to the name column when switching for now.

**Comment by PVince81** [comment]
- [ ] aria label for sort indicator
- [ ] own vue component for clickable sort column, there's already too much markup...

**Comment by PVince81** [comment]
I've added clickable columns in more lists, and also the whole width is clickable.
Since there's a lot of repetition, let's move those into a separate Vue component.

Also will need a way to detect those indicators from the test to find out whether to click once or twice depending on the desired sort order.

**Comment by PVince81** [comment]
- [ ] use raw oc-button instead of links

**Comment by PVince81** [comment]
- [ ] use raw oc-button instead of links, requires ODS update to not look shitty

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 1]
```diff
@@ -721,6 +721,29 @@ module.exports = {
       return indicators
     },
 
+    setSort: async function (column, isDesc = false) {
+      const columnSelector = util.format(this.elements.filesTableHeaderColumn.selector, xpathHelper.buildXpathLiteral(column))
+      await this.useXpath()
+        .waitForElementVisible(columnSelector)
```
Comment (LukasHirt) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js]: From nightwatch docs:

> click() will automatically wait for the element to be present (until the specified timeout).

Do we need to wait until the element is visible or is present enough?

Comment (PVince81) [reply tests/acceptance/pageObjects/FilesPageElement/filesList.js]: let's keep "visible" just to be on the safe side...

**Pull request merged**